xds/src/main/java/io/grpc/xds/internal/certprovider/CertificateProvider.java

```diff
@@ -47,33 +49,57 @@
 
   @VisibleForTesting
   static final class DistributorWatcher implements Watcher {
+    private PrivateKey lastKey;
+    private List<X509Certificate> lastCertChain;
+    private List<X509Certificate> lastTrustedRoots;
+
     @VisibleForTesting
     final Set<Watcher> downsstreamWatchers = new HashSet<>();
 
     synchronized void addWatcher(Watcher watcher) {
       downsstreamWatchers.add(watcher);
+      if (lastKey != null && lastCertChain != null) {
+        sendLastCertificateUpdate(watcher);
+      }
+      if (lastTrustedRoots != null) {
+        sendLastTrustedRootsUpdate(watcher);
+      }
     }
 
     synchronized void removeWatcher(Watcher watcher) {
       downsstreamWatchers.remove(watcher);
     }
 
+    private void sendLastCertificateUpdate(Watcher watcher) {
+      checkNotNull(lastKey, "lastKey");
+      checkNotNull(lastCertChain, "lastCertChain");
```

Comment:
Maybe move the null checks to the public API `updateCertificate(PrivateKey key, List<X509Certificate> certChain)`.

Reply:
done